### ISSUE TITLE
"when they occur"?

### ISSUE THREAD

**Issue by marcoscaceres** [sysapps/telephony#127]
The spec says: 

> Note that compliant implementations may not be reporting such events when they occur (e.g. OTA SIM update or hot-swappable SIM cards).

Does "when they occur" above mean that: 
1. service is added/removed, but no event may be fired at all?
2. service is added/removed, but the event is fired in the future?

**Comment by zolkis** [comment]
Yes, if the sim card is hot-swappable, the implementation has to poll for changes, therefore the event is fired in the future, which means there may be a period of time when it's uncertain if a serviceId denotes an enabled service. In this case, operations on the service will fail, so the client code will spend more time on error ramifications.
